rust/src/enhancers/matchers.rs

```diff
@@ -335,9 +335,44 @@ impl fmt::Display for FrameMatcherInner {
     }
 }
 
+impl PartialEq for FrameMatcherInner {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (
+                FrameMatcherInner::Field {
+                    field, path_like, ..
+                },
+                FrameMatcherInner::Field {
+                    field: other_field,
+                    path_like: other_path_like,
+                    ..
+                },
+            ) => field == other_field && path_like == other_path_like,
```

Comment:
You are not comparing the `pattern` here.
Now I wonder if that is even serialized properly?

I also see that `ExceptionMatcher` down below has both a `pattern` and `raw_pattern`. Maybe its worth wrapping that into a struct and `impl PartialEq` for that using the `raw_pattern`, and derive `PartialEq` for these two here instead.

Reply:
Actually reviewing this again, I wonder why we have `pattern` on `FrameMatcherInner` at all, if we only use the `raw_pattern` of the outer `FrameMatcher`?

Reply:
We do use the `pattern` for matching. I think for serialization the `raw_pattern` is the right choice. Unfortunately `bytes::Regex` is not `PartialEq` or this would all be much easier.

Reply:
Btw all these `PartialEq` shenanigans exist because of a test (I want to check if a roundtrip works), but we should probably be able to test matchers &c. for equality.